cg/cli/deliver/base.py

```diff
@@ -67,48 +67,21 @@ def deliver_analysis(
     force_all: bool,
     ignore_missing_bundles: bool,
 ):
-    """Deliver analysis files to customer inbox
-
+    """Deliver analysis files to customer inbox.
     Files can be delivered either on case level or for all cases connected to a ticket.
-    Any of those needs to be specified.
     """
     if not (case_id or ticket):
         LOG.info("Please provide a case-id or ticket-id")
         return
 
-    inbox: str = context.delivery_path
-    if not inbox:
-        LOG.info("Please specify the root path for where files should be delivered")
-        return
+    delivery_api: DeliveryAPI = context.delivery_api
+
+    delivery_api.set_dry_run(dry_run)
+    delivery_api.set_deliver_failed_samples(force_all)
+    delivery_api.set_ignore_missing_bundles(ignore_missing_bundles)
 
-    status_db: Store = context.status_db
     for delivery in delivery_type:
-        deliver_api = DeliverAPI(
-            store=status_db,
-            hk_api=context.housekeeper_api,
-            case_tags=PIPELINE_ANALYSIS_TAG_MAP[delivery]["case_tags"],
-            sample_tags=PIPELINE_ANALYSIS_TAG_MAP[delivery]["sample_tags"],
-            project_base_path=Path(inbox),
-            delivery_type=delivery,
-            force_all=force_all,
-            ignore_missing_bundles=ignore_missing_bundles,
-        )
-        deliver_api.set_dry_run(dry_run)
-        cases: list[Case] = []
-        if case_id:
-            case_obj: Case = status_db.get_case_by_internal_id(internal_id=case_id)
-            if not case_obj:
-                LOG.warning(f"Could not find case {case_id}")
-                return
-            cases.append(case_obj)
-        else:
-            cases: list[Case] = status_db.get_cases_by_ticket_id(ticket_id=ticket)
-            if not cases:
-                LOG.warning(f"Could not find cases for ticket {ticket}")
-                return
-
-        for case_obj in cases:
-            deliver_api.deliver_files(case_obj=case_obj)
+        delivery_api.deliver(case_id=case_id, ticket=ticket, pipeline=delivery)
```

Comment:
Another thought: Would you consider including the for loop inside the delivery API, similar to what is done in the demux post-processing API?

Reply:
I'll leave this to limit the scope of the refactoring